### PR TITLE
[SPELL] Vortex Wormholes

### PR DESCRIPTION
**Pull request by Time-Green**
Adds a new 2 spell points wizard spell.

It gives the wizard a hand held thing he must throw. It will then suck in all objects and people in a 7 tile radius, transporting them to a holding cell. The wizard is not effected by the pull and can enter and exit freely.

## Utilities:
Capture people
Collect objects
Shelter
Collect objects through walls 

## The storage?
I already know exactly what it is, but I'm not going to spoil that.
People can die inside if they are morons
They cannot enter and exit like the wizard
The wizard can reopen the portal to the last used location
Everyone in wizards pull can exit. People pulled by the wizards are no effected by the suction
There is a way to escape. You'll know how once you get in. It's not crazily difficult, everyone can do it if they really want.

Other non-vortex stuff

## Added a new summon event event
Telelockers (and crates)! It turns a few lockers into telelockers. They will be indestinguishable from normal lockers, other then that they teleport everything inside to another, not locked, locker.

## Tinker Tool
I left this out because there exists no amount of spell points that makes up for the power of this item.
It turns things into their subtypes, if they have them. You can turn an ID into captains ID and a laser into a gib gun. Yeah.

### CODE REVIEW

**Comment by Ratvar** [comment]
Change mining linked backpacks into linked telelockers.
Can you build things in pocket space? Relise gases? Fill with CO2, make turret-profit. 
What about sucking armory in one second?
What about filling pocket space with mimics?
Does radio works in pocket, or can you abduct and mindswap people stealthy?
Can summoned sublings enter/flee from pocket space? Place camera console/telecomm/pda/teleporter/protolathe/uploads here , get abductors 2.0
Oh, the possibilities.

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm [hunk 1]
```diff
@@ -105,3 +105,15 @@
...
+/obj/structure/closet/teleport
+	health = 99999
+	can_weld_shut = FALSE
+	storage_capacity = 10
+	cutting_tool = null
+	magic_teleport = TRUE
+	anchored = 1
+
+/obj/structure/closet/teleport/Destroy()
+	return
```
Comment (Cruix) [inline code/game/objects/structures/crates_lockers/closets/gimmick.dm]: All this will do is cause bugs, it will not stop the object from being deleted

File: code/modules/spells/spell_types/thrown.dm [hunk 2]
```diff
@@ -133,4 +133,179 @@
...
+	icon_state = icon_state_closed
+	var/i = 0
+	for(var/obj/item/I in orange(6,src))
+		if(i >= 20)
+			break
+		step(I,pick(NORTH,SOUTH,EAST,WEST))
+		i++
+
+/obj/effect/vortex_end/Bumped(mob/A)
+	if(iswizard(A) || iswizard(A.pulledby))
```
Comment (Cruix) [inline code/modules/spells/spell_types/thrown.dm]: Do what timestop does and make a list of immune mobs, don't just make all wizards immune to it.

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm (hunk 1 above)
Comment (Time-Green) [reply code/game/objects/structures/crates_lockers/closets/gimmick.dm]: Alright, I'll change it to use the same code as the vortex end, that's copy pasted from good code

File: code/modules/spells/spell_types/thrown.dm (hunk 2 above)
Comment (Time-Green) [reply code/modules/spells/spell_types/thrown.dm]: I'll just make it everyone that bought the spell

File: code/game/objects/structures/crates_lockers/closets/gimmick.dm (hunk 1 above)
Comment (Super3222) [reply code/game/objects/structures/crates_lockers/closets/gimmick.dm]: If it’s indestructible, might wanna make it ignore bomb damage. (Called by ex_act)

Comment (Time-Green) [reply code/game/objects/structures/crates_lockers/closets/gimmick.dm]: I fear people will find ways to destroy it, overriding qdel is the true way to get rid.
There's a way better proc to do this, I'll update it with that

File: code/game/gamemodes/wizard/artefact.dm [hunk 3]
```diff
@@ -755,3 +755,38 @@ var/global/list/multiverse = list()
 
 /obj/structure/table/abductor/wabbajack/right
 	desc = "It wakes so you may sleep."
+
+/obj/item/weapon/tinkertool
+	name = "tinker tool"
+	desc = "One of the most powerful artefacts. Turns things into other things."
```
Comment (Super3222) [inline code/game/gamemodes/wizard/artefact.dm]: artifacts!

File: code/modules/spells/spell_types/thrown.dm [hunk 4]
```diff
@@ -133,4 +133,179 @@
...
+	else if(isliving(A))
+		var/mob/living/L = A
+		L.Stun(4)
+		playsound(get_turf(src), 'sound/magic/LightningShock.ogg', 30, 1, -1)
+
+
+/obj/effect/vortex_end/attack_hand(mob/user)
+	if(iswizard(user) && opened)
+		for(var/obj/effect/vortex/vortex in vortexes)
+			if(vortex.creator = user)
```
Comment (Cruix) [inline code/modules/spells/spell_types/thrown.dm]: The compiler is complaining about this and it is probably right

File: code/game/gamemodes/wizard/artefact.dm [hunk 5]
```diff
@@ -755,3 +755,38 @@ var/global/list/multiverse = list()
 
 /obj/structure/table/abductor/wabbajack/right
 	desc = "It wakes so you may sleep."
+
+/obj/item/weapon/tinkertool
```
Comment (Cruix) [inline code/game/gamemodes/wizard/artefact.dm]: Just remove this altogether. It is unrelated to the rest of the PR, it can create abstract items that should never be instantiated, and anyone who is able to use this can just use spawn instead.

File: code/modules/spells/spell_types/wizard.dm [hunk 6]
```diff
@@ -338,7 +338,7 @@
 	qdel(targeteffect)
 
 /obj/effect/proc_holder/spell/aoe_turf/repulse/xeno //i fixed conflicts only to find out that this is in the WIZARD file instead of the xeno file?!
-	name = "Tail Sweep"
+	name = "Tail Sweep"                             //Then fucking move it, instead of crying about it like a little bitch
```
Comment (Cruix) [inline code/modules/spells/spell_types/wizard.dm]: Remove this comment

File: code/modules/spells/spell_types/wizard.dm [hunk 7]
```diff
@@ -356,3 +356,5 @@
 		playsound(C.loc, 'sound/voice/hiss5.ogg', 80, 1, 1)
 		C.spin(6,1)
 	..()
+
+/obj/effect/proc_holder/spell/targeted/astral_form
```
Comment (Cruix) [inline code/modules/spells/spell_types/wizard.dm]: Why is this here?

File: code/modules/spells/spell_types/thrown.dm (hunk 4 above)
Comment (Super3222) [reply code/modules/spells/spell_types/thrown.dm]: ==

File: code/modules/spells/spell_types/wizard.dm (hunk 7 above)
Comment (Time-Green) [reply code/modules/spells/spell_types/wizard.dm]: I was waiting for sprites from kmc, he never send them. Apparently I didnt remove all of it

**Comment by ghost** [comment]
>wizard version of ninja hell

exit

**Comment by Time-Green** [comment]
Unlike ninja hell, you can get out just fine. I made it out like 3 out of 5 times

**Comment by Ratvar** [comment]
Can you place your own machinery in Pocket Space? Can subwizard summonee enter and leave it at will?

**Comment by Time-Green** [comment]
You can place down machinery but it wont work. 
And like it says in my PR, yes wizards can exit and enter freely

**Comment by Mrcreepssss** [comment]
you should put a blacklist on that tinker tool. or a whitelist, depending on what's more fit.

**Comment by Time-Green** [comment]
Its been removed
